### ISSUE TITLE
Add optional redirection of stderr

### ISSUE THREAD

**Issue by jonathandturner** [nushell/nushell#2259]
**Is your feature request related to a problem? Please describe.**
We currently don't have a way to redirect or quiet stderr messages. We removed redirecting stderr because external commands should be able to rely on Nu, by default, behaving similar to other shells.

For some cases, though, being able to control if stderr is redirected will help us build more interesting commands in the future.

For example, I use a prompt pipeline that calls into the external `git`. Previously, if the directory wasn't a git directory, we could swallow the stderr message so the user would just see the prompt.

Now, it looks like this:

![Screenshot from 2020-07-26 08-03-01](https://user-images.githubusercontent.com/547158/88465398-da7bd480-cf16-11ea-9e5c-e5ada30a8a84.png)

**Describe the solution you'd like**
We should have a way to call into an external command and redirect any messages from stderr. 

For some reason, I was wondering if we wanted to make running externals an internal command and then leverage that. We currently have the `run_external` command, which is assumed to be a internal-to-Nu command.  I wonder if, instead, we should expose this command as `external`

```
external git branch `-l`
```
In this way, you could have more precise control over how an external is run. For example, let's say we gave it a flag to redirect stderr:

```
external --redirect-stderr git branch `-l`
```

The flag would work because of `external` being an internal command.

The one drawback of this approach is that for any flags passed to the external, you'd need to wrap them in quotes or backticks. Since largely the `external` command would be implied, I don't think this is necessarily a bad trade-off. And having the advantage of a configurable `external` seems pretty nice.

**Describe alternatives you've considered**
We could also put stderr redirection somewhere else. This would mean creating an alternative way of running externals, but it could also remain hidden.

**Comment by jonathandturner** [comment]
cc @thegedge

**Comment by thegedge** [comment]
Making it a command may come with difficulties, given how we parse externals differently. Another approach may be to take a block-based approach:

```
redirect_stderr { git status }
```

`redirect_stderr` would mutate the context in some way so that the external runner would know stderr should be redirected.

**Comment by jonathandturner** [comment]
@thegedge - along those lines, I wonder if the `is_last` feature we currently use is to help us know how to handle the redirects. I wonder if that should be an enum:

```
enum RedirectOutput {
  Stdout,
  Stderr,
  StdoutAndStderr,
  None,
}
```

This would let folks also redirect *just* stderr if they wanted.

**Comment by thegedge** [comment]
Yeah, perhaps I should have made that change instead, and just make it alway have been `Stdout`. That way the plumbing would have been in place for when we have a way to redirect stderr.

**Comment by CGMossa** [comment]
I'm here from the issue, that I've raised on discord regarding the fact that
`cargo run --example bla | save --raw log.txt` does not contain calls made by
`dbg!`-macro.
